### PR TITLE
fleetd: support conflict in global unit

### PR DESCRIPTION
**Pull request by dongsupark**
Support ``Conflict`` flag in global units, by allowing ``Conflict`` in metadata format, as well as allowing agent reconciler to check if a global unit has conflicts.
This PR is based on #1270 that was originally written by @wuqixuan, On top of that, I also implemented a small fix in agent reconciler, as well as a new functional test ``TestScheduleGlobalConflicts``.

Fixes: https://github.com/coreos/fleet/issues/1109
Supersedes: https://github.com/coreos/fleet/pull/1270

### CODE REVIEW

**Comment by ewoutp** [comment]
I would really like this PR merged. 
What is it waiting for?

**Comment by dongsupark** [comment]
@ewoutp Code review by someone else would really help, just like other pending PRs.
And I actually haven't thought this had a high priority.

**Comment by ewoutp** [comment]
I need this for the following setup.

- ServiceA (single instance)
- ServiceB (single instance on same host as ServiceA) (this I can already do)
- ServiceC (global on all hosts except the host of ServiceA) (this I need this PR for)

I'll test this PR here and let you know my findings.

File: agent/reconcile.go [hunk 1]
```diff
@@ -142,16 +142,25 @@ func desiredAgentState(a *Agent, reg registry.Registry) (*AgentState, error) {
 	for _, u := range units {
 		u := u
 		md := u.RequiredTargetMetadata()
-		if u.IsGlobal() && !machine.HasMetadata(&ms, md) {
```
Comment (ewoutp) [inline agent/reconcile.go]: Why this change? It does not seem to make a difference.

Comment (dongsupark) [inline agent/reconcile.go]: @ewoutp Exactly. It doesn't make any difference.
Why did it change? Short answer is, there's no particular reason.

Long answer is like that: @wuqixuan wrote the 1st patch ("fleetd: support conflict in global unit") by splitting the loop into 2 loops. However, @jonboulle [suggested](https://github.com/coreos/fleet/pull/1270#discussion_r40958036) keeping a single loop. Unfortunately the original author @wuqixuan didn't respond, so I took it over, and wrote the 3rd patch ("agent: squash loops into a single one in desiredAgentState") to follow @jonboulle's suggestion. The patch effectively changed it back to a single loop, but code style in that particular part changed slightly. Thus now you see the difference in a unified diff view.

I suppose this difference is not critical for the entire functionality. But if you want me to change it again, sure, I can update the patch again.

Comment (ewoutp) [inline agent/reconcile.go]: Got it, I would make more changes on it.

**Comment by ewoutp** [comment]
Code LGTM.

I've tried to test it but this was inconclusive. Unit tests are all ok. 
When testing this PR merged into master and running on my cluster (other fleet instances are v0.11.7) I got errors. They seem not to come from this PR because I get the same errors (etcd key already exists & nil-ptr panic) when testing master.

**Comment by dongsupark** [comment]
@ewoutp Thanks for the review. I'll merge it tomorrow.
BTW the other error sounds interesting, but that should be handled in other issue, I think.

**Comment by ewoutp** [comment]
@dongsupark Great.

For the other issue, see #1622

**Comment by dongsupark** [comment]
Merged. Thanks.
